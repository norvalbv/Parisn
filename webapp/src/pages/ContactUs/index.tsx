import { ReactElement } from 'react';
import useUser from '../../hooks/useUser';
import { DASHBOARD_IMAGE } from '../../constants';
import CardWrapper from '../../components/CardWrapper';
import Form from '../../components/Form';

const ContactUs = (): ReactElement => {
  const { user } = useUser();

  return (
    <div className="flex flex-row-reverse">
      <img src={DASHBOARD_IMAGE} alt={DASHBOARD_IMAGE} className="h-screen w-[42.5%]" />
      <CardWrapper cardType="centered">
        <div className="flex flex-col text-center gap-4 tracking-wider">
          <h2 className="text-4xl underline mx-auto relative">Contact Us</h2>
          <h3 className="my-6">Contact Parisn Customer Support</h3>
          <Form
            formValues={{
              firstName: { initialValue: user.id ?? '', type: 'text', label: 'First Name' },
              lastName: { initialValue: user.id ?? '', type: 'text', label: 'Last Name' },
              email: { initialValue: user.id ?? '', type: 'text', label: 'Email' },
              orderNumber: { initialValue: '', type: 'text', label: 'Order Number' },
              Message: { initialValue: '', type: 'textarea', label: 'Message' },
            }}
            submitButtonText="Submit Query"
          />
        </div>
      </CardWrapper>
    </div>
  );
};

export default ContactUs;
