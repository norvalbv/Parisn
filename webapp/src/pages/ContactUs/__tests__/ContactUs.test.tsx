import { FC } from 'react';
import renderer from 'react-test-renderer';

import ContactUs from '../';

describe('<ContactUs />', () => {
  test('ContactUs props with mandatory props', () => {
    const tree = renderer.create(<ContactUs />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
