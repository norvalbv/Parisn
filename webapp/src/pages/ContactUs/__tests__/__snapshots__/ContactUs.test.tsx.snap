// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ContactUs /> ContactUs props with mandatory props 1`] = `
<div>
  ContactUs
</div>
`;
